service/storage_service.cc

```diff
@@ -2928,21 +2928,23 @@ future<> storage_service::removenode_with_stream(gms::inet_address leaving_node,
 }
 
 future<> storage_service::restore_replica_count(inet_address endpoint, inet_address notify_endpoint) {
+    _abort_source.check();
+    auto sas = make_shared<abort_source>();
+    auto sub = _abort_source.subscribe([sas] () noexcept {
+        if (!sas->abort_requested()) {
+            sas->request_abort();
+        }
```

Comment:
If we wouldn't need the shared abort_source it would be possible.
That said, in the task_manager world, to be able to abort a particular node operation it will need its own abort source, so each top level task will probably subscribe to the "global" abort_source that's triggered on shutdown
and the subscription will request the per-task abort_source to abort (if not already requested).
In addition, the task::impl abort method would request abort on the per-task abort_source as well.

Cc @Deexie

Reply:
If we have multiple levels, then multiple abort_sources are reasonable.

Probably the inner one should live in the task itself.

Reply:
> If we have multiple levels, then multiple abort_sources are reasonable.
> 
> Probably the inner one should live in the task itself.

Right, it does.